shell/platform/fuchsia/flutter/vsync_waiter.cc

```diff
@@ -33,11 +36,29 @@ VsyncWaiter::VsyncWaiter(std::string debug_label,
     FireCallbackNow();
   };
 
+  // Generate a WeakPtrFactory for use with the UI thread. This does not need
+  // to wait on a latch because we only ever use the WeakPtrFactory on the UI
+  // thread so we have ordering guarantees (see ::AwaitVSync())
+  fml::TaskRunner::RunNowOrPostTask(
+      task_runners_.GetUITaskRunner(), fml::MakeCopyable([this]() mutable {
+        this->weak_factory_ui_ =
```

Comment:
@chinmaygarde should we block this call. I think its ok to not block but jut want to confirm.

Reply:
I think this is fine because of the reason mentioned in the comment.